### ISSUE TITLE
Some CLI lib triggers the IO stream twice on bun, which is different from node.js.

### ISSUE THREAD

**Issue by PrinOrange** [oven-sh/bun#9820]
### What version of Bun is running?

1.1.0+5903a6141

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

[inquirer.js](https://github.com/SBoudrias/Inquirer.js) is a **widely used basic library for command line interaction** , used to create interactive command line interfaces for asking and answering. 
It often serves as a foundational library and **is frequently relied upon as a dependency by other projects such as nest.js, Angular, etc. So the impact it has is widely.**

First, init a bun project
```bash
bun init
```
And install the inquirer.js
```bash
bun install inquirer
bun install @types/inquirer
```
Then make a simple usage for inquire.js, type such code in `index.ts`.
```typescript
import inquirer from 'inquirer';

const questions = [
  {
    type: 'input',
    name: 'name',
    message: 'Input your name',
  },
];

// Ask a question with inquire.js
inquirer.prompt(questions).then(answers => {
  const name = answers['name'];
  console.log(`Hello, ${name}, welcome!`);
});
```

Then run the script, **it will trigger the IO stream two times.**

https://github.com/oven-sh/bun/assets/73331790/ad60ac93-1a72-4ef0-8d7e-22c71e2a75f9

But on the contrary, if you use node.js to execute the above same code, there will be no such bug.

https://github.com/oven-sh/bun/assets/73331790/713d98f1-6c9a-4956-8012-298e5827c452

**Comment by sohaieb** [comment]
<ins>I also confirm the same issue</ins>. 
**Furthermore**, the `@prompt/select` does not work well when using the keyboard (arrows up/bottom) to chose an option

**Comment by PrinOrange** [comment]
This bug was solved in 1.1.9
